openapi_python_client/parser/properties.py

```diff
@@ -222,7 +224,7 @@ class ListProperty(Property, Generic[InnerProp]):
 
     def get_type_string(self, no_optional: bool = False) -> str:
         """ Get a string representation of type that should be used when declaring this property """
-        if no_optional or (self.required and not self.nullable):
+        if no_optional or not self.nullable:
             return f"List[{self.inner_property.get_type_string()}]"
         return f"Optional[List[{self.inner_property.get_type_string()}]]"
```

Comment:
Probably need something like this, and I expect similar in other places. The good news is mypy should yell at us if we get some of these wrong (assuming full e2e coverage),

```suggestion
        type_string = f"List[{self.inner_property.get_type_string()}]"
        if not no_optional and self.nullable:
            type_string = f"Optional[{type_string}]" 
        if not self.required:
            type_string = f"Union[Unset, {type_string}]"
        return type_string
```